### PR TITLE
configurable-network

### PR DESCRIPTION
**Pull request by calvix**
adding mayu option for setting subnet length and vlan id

see https://github.com/giantswarm/hive/pull/21 and https://github.com/giantswarm/hive/issues/20

### CODE REVIEW

File: pxemgr/config.go [hunk 1]
```diff
@@ -66,6 +66,8 @@ type Network struct {
 	Router       string
 	DNS          []string
 	PXE          bool
+	Subnet       string `yaml:"subnet_len"`
```
Comment (r7vme) [inline pxemgr/config.go]: SubnetSize and subnet_size

Comment (xh3b4sd) [inline pxemgr/config.go]: I would also like to keep the struct property and the JSON/YAML tags in sync.

**Comment by calvix** [comment]
I have updated variable names accordingly

File: template_snippets/cloudconfig/net_singlenic.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@
       Name={{.Host.ConnectedNIC}}
 
       [Network]
-      Address={{.Host.InternalAddr}}/22
+      Address={{.Host.InternalAddr}}/{{.ClusterNetwork.Subnet}}
```
Comment (r7vme) [inline template_snippets/cloudconfig/net_singlenic.yaml]: SubnetSize

Comment (calvix) [inline template_snippets/cloudconfig/net_singlenic.yaml]: fixed

**Pull request merged**